src/tokens/selectors.ts

```diff
@@ -490,3 +490,23 @@ export const feeCurrenciesWithPositiveBalancesSelector = createSelector(
 
 export const visualizeNFTsEnabledInHomeAssetsPageSelector = (state: RootState) =>
   state.app.visualizeNFTsEnabledInHomeAssetsPage
+
+export const tokenPriceHistorySelector = createSelector(
+  [
+    (state: RootState) => state.tokens.tokenBalances,
+    (_state: RootState, tokenId: string) => tokenId,
+  ],
+  (tokenBalances, tokenId) => {
+    return tokenBalances[tokenId]?.historicalPricesUsd?.priceHistory ?? []
+  }
+)
+
+export const tokenPriceHistoryStatusSelector = createSelector(
+  [
+    (state: RootState) => state.tokens.tokenBalances,
+    (_state: RootState, tokenId: string) => tokenId,
+  ],
+  (tokenBalances, tokenId) => {
+    return tokenBalances[tokenId]?.historicalPricesUsd?.priceHistoryStatus ?? 'idle'
```

Comment:
If `undefined` is the exact same as `idle` then there isn't much use for the idle state. I think either make priceHistoryStatus required and set the status to `idle` on the initial data OR remove the `idle` state. I have a slight preference for making it required so its absolutely clear.